weepickle-implicits/src/main/scala/com/rallyhealth/weepickle/v1/implicits/DefaultFroms.scala

```diff
@@ -158,12 +153,12 @@ trait DefaultFroms
 /**
   * This needs to be split into a separate trait due to https://github.com/scala/bug/issues/11768
   */
-trait LowPriFroms extends com.rallyhealth.weepickle.v1.core.Types {
+trait LowPriFroms { this: Types =>
   implicit def SeqLikeFrom[C[_] <: Iterable[_], T](implicit r: From[T]): From[C[T]] =
     new From[C[T]] {
       def transform0[R](v: C[T], out: Visitor[_, R]): R = {
         val ctx = out.visitArray(v.size).narrow
-        v.asInstanceOf[Iterable[T]].foreach { t =>
+        v.iterator.foreach { t =>
```

Comment:
```suggestion
        v..asInstanceOf[Iterable[T]].foreach { t =>
```
seems to work as is. `Iterable.foreach` might be optimized.

Reply:
Done. I just did a "clean up" commit to remove unnecessary changes like these. Also unnecessary reformatting. Also brought back`def`s instead of `val`s in `RW` definitions, since recursion issues are addressed.